Index: h323plus/plugins/video/H.263-ffmpeg/h263ffmpeg.cxx
===================================================================
--- h323plus.orig/plugins/video/H.263-ffmpeg/h263ffmpeg.cxx
+++ h323plus/plugins/video/H.263-ffmpeg/h263ffmpeg.cxx
@@ -84,12 +84,10 @@
 
 
 extern "C" {
-#include "ffmpeg/avcodec.h"
+#include <libavcodec/avcodec.h>
 };
 
-#if LIBAVCODEC_VERSION_INT != 0x000406
-#error Wrong libavcodec version for h.263.
-#endif
+#define FULL_LAVC_LIB_NAME "libavcodec.so." #LIBAVCODEC_VERSION_MAJOR
 
 #  ifdef  _WIN32
 #    define P_DEFAULT_PLUGIN_DIR "C:\\PTLIB_PLUGINS;C:\\PWLIB_PLUGINS"
@@ -395,10 +393,7 @@ class FFMPEGLibrary
 
   protected:
     void (*Favcodec_init)(void);
-    AVCodec *Favcodec_h263_encoder;
-    AVCodec *Favcodec_h263p_encoder;
-    AVCodec *Favcodec_h263_decoder;
-    void (*Favcodec_register)(AVCodec *format);
+    void (*Favcodec_register_all)(void);
     AVCodec *(*Favcodec_find_encoder)(enum CodecID id);
     AVCodec *(*Favcodec_find_decoder)(enum CodecID id);
     AVCodecContext *(*Favcodec_alloc_context)(void);
@@ -407,7 +402,7 @@ class FFMPEGLibrary
     int (*Favcodec_open)(AVCodecContext *ctx, AVCodec *codec);
     int (*Favcodec_close)(AVCodecContext *ctx);
     int (*Favcodec_encode_video)(AVCodecContext *ctx, BYTE *buf, int buf_size, const AVFrame *pict);
-    int (*Favcodec_decode_video)(AVCodecContext *ctx, AVFrame *pict, int *got_picture_ptr, BYTE *buf, int buf_size);
+    int (*Favcodec_decode_video2)(AVCodecContext *ctx, AVFrame *pict, int *got_picture_ptr, const AVPacket *avpkt);
 
     void (*Favcodec_set_print_fn)(void (*print_fn)(char *));
     unsigned (*Favcodec_version)(void);
@@ -433,11 +428,11 @@ bool FFMPEGLibrary::Load()
   if (IsLoaded())
     return true;
 
-  if (!DynaLink::Open("avcodec")
+  if (
 #if defined(_WIN32)
-      && !DynaLink::Open("libavcodec")
+      !DynaLink::Open("libavcodec")
 #else
-      && !DynaLink::Open("libavcodec.so")
+      !DynaLink::Open(FULL_LAVC_LIB_NAME)
 #endif
     ) {
     //cerr << "FFLINK\tFailed to load a library, some codecs won't operate correctly;" << endl;
@@ -450,27 +445,12 @@ bool FFMPEGLibrary::Load()
   }
 
   if (!GetFunction("avcodec_init", (Function &)Favcodec_init)) {
-    //cerr << "Failed to load avcodec_int" << endl;
-    return false;
-  }
-
-  if (!GetFunction("h263_encoder", (Function &)Favcodec_h263_encoder)) {
-    //cerr << "Failed to load h263_encoder" << endl;
-    return false;
-  }
-
-  if (!GetFunction("h263p_encoder", (Function &)Favcodec_h263p_encoder)) {
-    //cerr << "Failed to load h263p_encoder" << endl;
-    return false;
-  }
-
-  if (!GetFunction("h263_decoder", (Function &)Favcodec_h263_decoder)) {
-    //cerr << "Failed to load h263_decoder" << endl;
+    //cerr << "Failed to load avcodec_init" << endl;
     return false;
   }
 
-  if (!GetFunction("register_avcodec", (Function &)Favcodec_register)) {
-    //cerr << "Failed to load register_avcodec" << endl;
+  if (!GetFunction("avcodec_register_all", (Function &)Favcodec_register_all)) {
+    //cerr << "Failed to load avcodec_register_all" << endl;
     return false;
   }
 
@@ -509,8 +489,8 @@ bool FFMPEGLibrary::Load()
     return false;
   }
 
-  if (!GetFunction("avcodec_decode_video", (Function &)Favcodec_decode_video)) {
-    //cerr << "Failed to load avcodec_decode_video" << endl;
+  if (!GetFunction("avcodec_decode_video2", (Function &)Favcodec_decode_video2)) {
+    //cerr << "Failed to load avcodec_decode_video2" << endl;
     return false;
   }
 
@@ -550,11 +530,8 @@ bool FFMPEGLibrary::Load()
   // must be called before using avcodec lib
   Favcodec_init();
 
-  // register only the codecs needed (to have smaller code)
-  Favcodec_register(Favcodec_h263_encoder);
-  Favcodec_register(Favcodec_h263p_encoder);
-  Favcodec_register(Favcodec_h263_decoder);
-  
+  Favcodec_register_all();
+
   //Favcodec_set_print_fn(h263_ffmpeg_printon);
 
   isLoadedOK = true;
@@ -627,7 +604,10 @@ int FFMPEGLibrary::AvcodecDecodeVideo(AV
 
   //PTRACE(6, "FFLINK\tNow decode video for ctxt @ " << ::hex << (int)ctx << ", pict @ " << (int)pict
 	// << ", buf @ " << (int)buf << ::dec << " (" << buf_size << " bytes)");
-  int res = Favcodec_decode_video(ctx, pict, got_picture_ptr, buf, buf_size);
+  AVPacket av_packet = {};
+  av_packet.data = buf;
+  av_packet.size = buf_size;
+  int res = Favcodec_decode_video2(ctx, pict, got_picture_ptr, &av_packet);
 
   //PTRACE(6, "FFLINK\tDecoded video of " << res << " bytes, got_picture=" << *got_picture_ptr);
   return res;
@@ -967,7 +947,9 @@ bool H263EncoderContext::OpenCodec()
 
   avcontext->flags |= CODEC_FLAG_RFC2190;
 
+#ifdef LIBAVCODEC_RTP_MODE
   avcontext->rtp_mode = 1;
+#endif
   avcontext->rtp_payload_size = 1400;
   avcontext->rtp_callback = &H263EncoderContext::RtpCallback;
   avcontext->opaque = this; // used to separate out packets from different encode threads
Index: h323plus/plugins/configure.ac
===================================================================
--- h323plus.orig/plugins/configure.ac
+++ h323plus/plugins/configure.ac
@@ -260,10 +260,10 @@ else
                                            HAVE_H264_DECODER=yes
                                            ;;
                                        * ) 
-                                           AC_CHECK_LIB(avcodec, h263p_encoder, [HAVE_H263=yes], [HAVE_H263=no])
-                                           AC_CHECK_LIB(avcodec, h263p_encoder, [HAVE_H263P=yes], [HAVE_H263P=no])
-                                           AC_CHECK_LIB(avcodec, mpeg4_encoder, [HAVE_MPEG4=yes], [HAVE_MPEG4=no])
-                                           AC_CHECK_LIB(avcodec, h264_decoder, [HAVE_H264_DECODER=yes], [HAVE_H264_DECODER=no])
+                                           HAVE_H263=yes
+                                           HAVE_H263P=yes
+                                           HAVE_MPEG4=yes
+                                           HAVE_H264_DECODER=yes
                                            ;;
                      esac
                      AC_SUBST(HAVE_H263)
Index: h323plus/plugins/video/H.263-ffmpeg/Makefile.in
===================================================================
--- h323plus.orig/plugins/video/H.263-ffmpeg/Makefile.in
+++ h323plus/plugins/video/H.263-ffmpeg/Makefile.in
@@ -62,6 +62,11 @@ LDSO	 	=@LDSO@
 PLUGINEXT	=@PLUGINEXT@
 STDCCFLAGS	=@STDCCFLAGS@
 LDFLAGS		=@LDFLAGS@
+HAVE_LIBAVCODEC_RTP_MODE=@HAVE_LIBAVCODEC_RTP_MODE@
+
+ifdef HAVE_LIBAVCODEC_RTP_MODE
+STDCCFLAGS += -DLIBAVCODEC_RTP_MODE=1
+endif
 
 EXTRACCFLAGS    += -I$(COMMONDIR) -I$(PLUGINDIR)
 
