Index: webrtc/src/webrtc/api/BUILD.gn
===================================================================
--- webrtc.orig/src/webrtc/api/BUILD.gn	2017-01-12 16:02:32.837533588 +0300
+++ webrtc/src/webrtc/api/BUILD.gn	2017-01-12 16:02:32.835533598 +0300
@@ -24,7 +24,7 @@ group("api") {
   }
 }
 
-rtc_source_set("call_api") {
+rtc_static_library("call_api") {
   sources = [
     "call/audio_receive_stream.h",
     "call/audio_send_stream.cc",
