diff --git a/src/ptlib/unix/tlibthrd.cxx b/src/ptlib/unix/tlibthrd.cxx
index 42de3f3..cff3050 100644
--- a/src/ptlib/unix/tlibthrd.cxx
+++ b/src/ptlib/unix/tlibthrd.cxx
@@ -1533,7 +1533,11 @@ PTimedMutex::~PTimedMutex()
     for (PINDEX i = 0; i < 100; ++i) {
       if ((result = pthread_mutex_destroy(&m_mutex)) != EBUSY)
         break;
+#ifdef __MACH__
+	sched_yield();
+#else
       pthread_yield();
+#endif
     }
   }
 
