--- a/src.orig/ptlib/unix/svcproc.cxx	2016-11-08 21:07:18.000000000 +0300
+++ b/src/ptlib/unix/svcproc.cxx	2016-11-08 21:11:27.000000000 +0300
@@ -279,7 +279,7 @@
     pid_t pid;

     {
-      ifstream pidfile(pidfilename);
+	  ifstream pidfile(static_cast<const char*>(pidfilename));
       if (!pidfile.is_open()) {
         cout << "Could not open pid file: \"" << pidfilename << "\""
                 " - " << strerror(errno) << endl;
@@ -422,7 +422,7 @@
   // Run as a daemon, ie fork

   if (!pidfilename) {
-    ifstream pidfile(pidfilename);
+	ifstream pidfile(static_cast<const char*>(pidfilename));
     if (pidfile.is_open()) {
       pid_t pid;
       pidfile >> pid;
@@ -450,7 +450,7 @@
       cout << "Daemon started with pid " << pid << endl;
       if (!pidfilename) {
         // Write out the child pid to magic file in /var/run (at least for linux)
-        ofstream pidfile(pidfilename);
+        ofstream pidfile(static_cast<const char*>(pidfilename));
         if (pidfile.is_open())
           pidfile << pid;
         else
