diff --git a/configure.ac b/configure.ac
index dd59e27..67fc0cd 100644
--- a/configure.ac
+++ b/configure.ac
@@ -969,7 +969,7 @@ AC_PROG_INSTALL
 
 AC_CONFIG_FILES(openh323u.mak)
 AC_CONFIG_FILES(Makefile)
-
+AC_CONFIG_FILES(h323plus.pc)
 
 dnl ########################################################################
 dnl Header file
@@ -983,4 +983,3 @@ dnl Output configured files
 AC_CONFIG_SUBDIRS(plugins)
 
 AC_OUTPUT()
-
