Author: Tzafrir Cohen <tzafrir@debian.org>
Description: minor fixes to fix newer versions of autoconf
Index: h323plus/plugins/configure.ac
===================================================================
--- h323plus.orig/plugins/configure.ac
+++ h323plus/plugins/configure.ac
@@ -474,7 +474,7 @@ AC_COMPILE_IFELSE([
                   )
                   ],
   AC_MSG_RESULT(yes)
-  AC_DEFINE(HAVE_POSIX_MEMALIGN, 1),
+  AC_DEFINE(HAVE_POSIX_MEMALIGN, 1, [POSIX memalign() support]),
   AC_MSG_RESULT(no)
 )
 AC_LANG_POP([C++])
