Author: Kilian Krause <kilian@debian.org>
Description: Remove -DPASN_NOPRINTON from the CCFLAGS.
Index: h323plus/openh323u.mak.in
===================================================================
--- h323plus.orig/openh323u.mak.in
+++ h323plus/openh323u.mak.in
@@ -112,7 +112,7 @@ OH323_LIBDIR = $(OPENH323DIR)/lib
 endif # OH323_LIBDIR
 
 ifeq ($(NOTRACE), 1)
-STDCCFLAGS += -DPASN_NOPRINTON -DPASN_LEANANDMEAN
+STDCCFLAGS += -DPASN_LEANANDMEAN
 OH323_SUFFIX = n
 else
 ifeq (,$(findstring PTRACING,$(STDCCFLAGS)))
