--- h323plus/openh323u.mak.in.orig	2017-10-09 15:53:39.000000000 +0300
+++ h323plus/openh323u.mak.in	2017-10-09 15:53:57.000000000 +0300
@@ -64,7 +64,7 @@
 ifndef HAS_PTLIB_LIB_MAK
 
 ##### grab ptlib build settings
-include $(PTLIBDIR)/make/ptbuildopts.mak
+include $(PTLIBDIR)/make/ptlib_config.mak
 
 # compatibility with new ptlib - it's not has OPENSSL flag which cause undefined reference to `PPlugin_H235Authenticator_Std1_link()' in library
 ifdef HAS_SASL2
