Index: libg7221/configure.ac
===================================================================
--- libg7221.orig/configure.ac	2016-11-24 14:56:42.121063755 +0300
+++ libg7221/configure.ac	2016-11-24 14:56:42.117063767 +0300
@@ -239,10 +239,10 @@ gnu)
         COMP_VENDOR_CFLAGS="-msse4a $COMP_VENDOR_CFLAGS"
     fi
     if test "$enable_sse4_2" = "yes" ; then
-        COMP_VENDOR_CFLAGS="-msse42 $COMP_VENDOR_CFLAGS"
+        COMP_VENDOR_CFLAGS="-msse4.2 $COMP_VENDOR_CFLAGS"
     fi
     if test "$enable_sse4_1" = "yes" ; then
-        COMP_VENDOR_CFLAGS="-msse41 $COMP_VENDOR_CFLAGS"
+        COMP_VENDOR_CFLAGS="-msse4.1 $COMP_VENDOR_CFLAGS"
     fi
     if test "$enable_sse3" = "yes" ; then
         COMP_VENDOR_CFLAGS="-msse3 $COMP_VENDOR_CFLAGS"
